src/test/java/com/keboola/tableexporter/TestConnectionTest.java

```diff
@@ -79,8 +66,44 @@ public void testInvalidConnection() throws IOException, URISyntaxException, Appl
 
         String[] args = {"testConnection", tmpFile};
 
-        exit.expectSystemExitWithStatus(1);
-        app.main(args);
+        try {
+            Application.main(args);
+        } catch (ExitException exception) {
+            assertEquals(1, exception.status);
+        }
+    }
+
+    @Test
+    public void testProxyUserInvalid() throws IOException {
+        JSONObject baseObj = new JSONObject();
+        JSONObject paramsObj = new JSONObject();
+        JSONObject dbNode = getDbJsonNode();
+        dbNode.put("proxyUser", "invalid-proxy-user");
+        paramsObj.put("db", dbNode);
+        baseObj.put("parameters", paramsObj);
+
+        File outputConfigFile = File.createTempFile("config", ".json");
+        String tmpFile = outputConfigFile.getAbsolutePath();
+        writeJsonConfigToFile(baseObj, tmpFile);
+
+        String[] args = {"testConnection", tmpFile};
+
+        try {
+            Application.main(args);
+            fail("Exception expected");
+        } catch (ExitException exception) {
+            assertEquals(1, exception.status);
+
+            String expectedLog = "executing action testConnection\n" +
+                    "Processing configuration file " + tmpFile + "\n" +
+                    "Connecting user system to database XE at oracle on port 1521\n" +
+                    "Proxy user = \"invalid-proxy-user\"\n";
+            assertEquals(expectedLog, systemOutRule.getLog());
+
+            // We cannot test it better: "feature not enabled: Enterprise User Security"
+            String expectedErrorLog = "Proxy user error: ORA-00439: feature not enabled: Enterprise User Security\n\n";
+            assertEquals(expectedErrorLog, systemErrRule.getLog());
```

Comment:
A hlavne aj ked program zavola exit, ak si mozem skontrolovat co zapisal na STDOUT/STDERR.

Reply:
Mame limitovany oracle, takze mozem otestovat iba pripad, ze to spadne s `feature not enabled: Enterprise User Security`.